### DISCUSSION TITLE
Will the hash value of the output file be the same every time the generated file is executed

### DISCUSSION THREAD

**Discussion by 123xyl** [openmopac/mopac#141] [Q&A]
Does the content of the output file of each test have to be the same except for the time inside? If you delete the time inside the .out file and then calculate its sha256sum value, will the sha256sum value be the same every time the generated file is generated?

**Comment by 123xyl** [comment]
How to judge the consistency and accuracy of MOPAC tests

**Comment by godotalgorithm** [comment]
Unfortunately, no. Besides time and date stamps, the numerical behavior of MOPAC will change a small amount depending on compiler, compiler flags, and choice of BLAS/LAPACK library. None of these things promise reproducible floating-point arithmetic to all decimal digits. Many iterative methods in MOPAC can amplify small numerical changes into larger effects in rare cases.

**Reply by 123xyl** [reply]
That is, the data in the file generated after each test is not necessarily the same as the previous one, right?

**Reply by godotalgorithm** [reply]
MOPAC doesn't have random components. If you run the same executable on the same input file, it should have the same output each time except for the time and date stamps. The numerical results can change slightly only when the executable is recompiled under different conditions.
